### DISCUSSION TITLE
Download an executable

### DISCUSSION THREAD

**Discussion by spartan27244** [CMSgov/price-transparency-guide-validator#67] [Ideas]
I want to be able to download an executable to validate the files. I have no idea what these are nor do I want them.
C++
Node (version 16.x)
NPM (version 8.5.x)
Docker (version 19.x)

**Comment by dgolden** [comment]
@spartan27244 To run the validator, you need to set up a suitable environment which includes those components. This is a developer task. Scott's instructions are solid, and I was able to build and run the validator on an AWS Linux VM.  

If you don't mind my asking, what is your goal in running the validator? Have you considered outsourcing that task to a vendor rather than doing it yourself?

**Comment by spartan27244** [comment]
My goal running the validator is to validate files received. We are a development shop, but do not use C++, Node, Dapper, etc. and seems a bit of a stretch just to get one executable to validate a file or 12.

**Comment by shaselton-usds** [comment]
@spartan27244  -- You don't need to use C++ directly, that is the point of leveraging docker. We are using docker because we didn't want to create and maintain an executable for each ISA.
